lib/chat_api/conversations.ex

```diff
@@ -223,6 +223,14 @@ defmodule ChatApi.Conversations do
     archive_conversation(conversation)
   end
 
+  def archive_conversations(days_closed \\ 14) do
```

Comment:
i'm a little worried that it's not obvious what this method is doing just by looking at it -- for example, if i'm reading the code and i see `Conversations.archive_conversations()`, i might think ALL conversations would be archived

i wonder if it would be better to be a bit more explicit about what's going on here... maybe something like
```elixir
def query_conversations_closed_for(days: days) do
  Conversation
  |> where([c], is_nil(c.archived_at))
  |> where(status: "closed")
  |> where([c], c.updated_at < ago(^days, "day"))
end

def archive_conversations(query) do
  Repo.update_all(query, set: [archived_at: DateTime.utc_now()])
end
```
and then we would do something like
```elixir
{n, nil} =
  Conversations.query_conversations_closed_for(days: 14)
  |> Conversations.archive_conversations()
```

what do you think?

Reply:
we could also have a method called something like `Conversations.archive_conversations_closed_for(days: days)`, i just want to make sure it's explicit which conversations are being archived :)

Reply:
@reichert621 :D
I like it, not only for being more explicit but also for being able to compose it more loosely with other querys for example :+1: